Question about the Raspberry pi example

I have followed the guide and now i have the deployments running but the pod keeps crashing. When i check the actual container the logs says: 

I0613 14:02:34.782924       1 light_mapper.go:168] Changing the state of the device to online
I0613 14:02:34.784542       1 light_mapper.go:242] Watching on the device twin values for device: led-light-instance-01
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x66e450]

goroutine 1 [running]:
main.equateTwinValue(0x0, 0x0, 0x0, 0x400007d2f0)
	/var/opt/godir/src/github.com/kubeedge/examples/led-raspberrypi/light_mapper.go:248 +0x198
main.main()
	/var/opt/godir/src/github.com/kubeedge/examples/led-raspberrypi/light_mapper.go:280 +0x16c

--- a/led-raspberrypi/light_mapper.go
+++ b/led-raspberrypi/light_mapper.go
@@ -244,7 +244,7 @@ func equateTwinValue(updateMessage DeviceTwinUpdate) {
 	go subscribe()
 	getTwin(updateMessage)
 	wg.Wait()
-	if deviceTwinResult.Twin[powerStatus].Actual == nil || *deviceTwinResult.Twin[powerStatus].Expected.Value != *deviceTwinResult.Twin[powerStatus].Actual.Value {
+	if deviceTwinResult.Twin[powerStatus].Expected != nil && ((deviceTwinResult.Twin[powerStatus].Actual == nil) && deviceTwinResult.Twin[powerStatus].Expected != nil || (*deviceTwinResult.Twin[powerStatus].Expected.Value != *deviceTwinResult.Twin[powerStatus].Actual.Value)) {
 		glog.Info("Expected Value : ", *deviceTwinResult.Twin[powerStatus].Expected.Value)
 		if deviceTwinResult.Twin[powerStatus].Actual == nil {
 			glog.Info("Actual Value: ", deviceTwinResult.Twin[powerStatus].Actual)

--- a/led-raspberrypi/sample-crds/led-light-device-instance.yaml
+++ b/led-raspberrypi/sample-crds/led-light-device-instance.yaml
@@ -18,4 +18,8 @@ spec:
 status:
   twins:
     - propertyName: power-status
+      desired:
+        metadata:
+          type: string
+        value: 'OFF'
 

